[WordPress → Jetpack app branding] Update Get Apps section

SiobhyB opened the pull request:
_Context:_ Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack.

#### Proposed Changes

This PR update references to the mobile app found on the Get Apps screen (found via _Me_ → _Get Apps_ in Calypso). The design has also be refreshed. 

<details>
 <summary>Here's a high-level overview of the code changes ⤵️</summary>

* The `jetpack/app-branding` feature flag has been used to limit the changes to Calypso's staging site, for now. Note, in the future, this feature flag will be removed along with all Jetpack-specific class names and WordPress-app-specific code. 
* `me/sidebar/index.jsx`: The sidebar has been updated to use the Jetpack icon from the [list of available Gridicons](http://automattic.github.io/gridicons/).
* `get-apps/index.jsx`: A body class has been added to enable the setting of a white background, plus the removal of extra default padding and other styles that don't match our desired design.
* For setting the two different background colours, the same CSS pattern was used as the one found for the WooCommerce landing page (see _My Sites_ → _WooCommerce_ for an example).
* A `getJetpackBrandedPanel()` function was created, returning the newly designed components, in both `get-apps/mobile-download-card.jsx` and `get-apps/desktop-download-card.jsx`. 
</details>

#### Screenshots

<details>
 <summary>Before/After (Desktop) ⤵️</summary>

Screenshots taken on a MacBook Pro using Google Chrome:

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/199476581-51273855-5c05-4a98-bc88-acf8b52ad3ed.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/199476642-2b53e3b5-b59d-4bb5-8cf5-ca8aaa2f9375.png" width="100%"> |

</details>

<details>
 <summary>Before/After (Mobile) ⤵️</summary>

Screenshots taken from an iPhone XR emulator using Chrome:

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/199477359-bf44a6ae-d18b-4af6-9e9d-b417c62c2033.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/199477206-8bf18ced-1625-4716-a253-777271fd051f.png" width="100%"> |

</details>

#### Testing Instructions

* Navigate to [the auto-generated live link](https://github.com/Automattic/wp-calypso/pull/69674#issuecomment-1299360337) then head to _Me_ → _Apps_. 
* Verify that the design displays as expected across different devices, colour schemes, etc.
* If testing on a development branch, it'd be helpful to verify that the older/current design (with WordPress app references) displays as expected when the `jetpack/app-branding` feature flag is disabled.  

#### Future Iterations

The following additional changes were included in the initial designs. I plan to address these in separate PRs, as they require additional changes:

* Add second Mac button for Apple Silicon.
* Add email input to retrieve a desktop download link from a mobile device.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

SiobhyB committed:
Update apps link in sidebar with Jetpack icon

With this commit, the apps link in the sidebar of Calypso's 'Me' section is updated. It now displays the Jetpack icon and an updated label. This change can only be seen when the 'jetpack/app-branding' flag is active.

SiobhyB committed:
Apply main layout styles e.g. padding, colors...

This commit applies the main layout styles that are going to be used with the redesign of the Get Apps section. This includes applying different background colors, padding, and ordering of subpanels when the 'jetpack/app-branding' flag is enabled.

SiobhyB committed:
Mobile section: Apply Jetpack-specific styles

SiobhyB committed:
Mobile section: Separate device-specific panels

With this commit, separate panels are added for users viewing on a mobile or a desktop device. This allows us to display appropriate options for each device e.g. a QR code is only displayed on non-mobile devices.

SiobhyB committed:
Desktop section: Apply Jetpack-specific styles

SiobhyB committed:
Add RTL support to animated icon

SiobhyB committed:
Remove unused prop

SiobhyB committed:
Refine function names for clarity

SiobhyB committed:
Update function names for consistency

SiobhyB commented:
👋 @paulforgione, I'd appreciate any feedback from a design perspective on this first pass. :) You can see the updates via the live branch link [here](https://github.com/Automattic/wp-calypso/pull/69674#issuecomment-1299360337). I'd appreciate any feedback from a design perspective. Some things I'm not certain about:

* In the desktop app section, I used the same icon across both desktop and mobile devices, though the designs show different icons. Should I change this to use the different version of the WordPress icon for desktop (as per the design)?

* I know there was some discussion related to the mobile app section's copy. Should this be changed?

* The font-sizes don't all precisely match up to the designs. In cases where the font-size in the designs didn't have a matching variable from this list, I went for the closest match:

https://github.com/Automattic/wp-calypso/blob/e054ff394bdf31cbc609e16376a3212a7c889b08/packages/typography/styles/variables.scss#L38-L48

Let me know what you think of the above as well as if there are any other changes that are needed. (Also, as per the "notes" section, there are a couple of other changes in the designs that I plan to work on in future PRs.)

dcalhoun commented on client/blocks/get-apps/desktop-download-card.jsx:
```diff
@@ -53,32 +61,62 @@ class DesktopDownloadCard extends Component {
 	getTranslateComponents( platform ) {
 		switch ( platform ) {
 			case 'MacIntel':
 				return {
 					firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
 					secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
 					thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
+					...( displayJetpackAppBranding && {
+						firstAvailableIcon: this.getPlatformImage( WINDOWS_LINK ),
+						secondAvailableIcon: this.getPlatformImage( LINUX_TAR_LINK ),
+						thirdAvailableIcon: this.getPlatformImage( LINUX_DEB_LINK ),
+					} ),
 				};
```
Nit: rather than mutating every property in object, WDYT about returning a different object? IMO it may be little easier to comprehend and, eventually, remove the conditional. If we choose to make this modification, we should apply it to the other object spreads in this function as well. 

```suggestion
				return displayJetpackAppBranding ? {
          firstAvailableIcon: this.getPlatformImage( WINDOWS_LINK ),
          secondAvailableIcon: this.getPlatformImage( LINUX_TAR_LINK ),
          thirdAvailableIcon: this.getPlatformImage( LINUX_DEB_LINK ),
        } : {
					firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
					secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
					thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
				};
```

SiobhyB replied:
I appreciate your perspective here. Zooming out, I can see how the mutation would be hard to parse. I've gone ahead to create separate objects in https://github.com/Automattic/wp-calypso/pull/69674/commits/59edabb39eac9a83792917b4e052316b30742863, as per your suggestion.

dcalhoun commented on client/blocks/get-apps/desktop-download-card.jsx:
```diff
@@ -153,22 +229,122 @@ class DesktopDownloadCard extends Component {
+		const { isMobile } = userAgent;
+
+		return (
+			<>
```
Nit: this Fragment appears unnecessary since there is a single child.

SiobhyB replied:
Good catch! Fixed in https://github.com/Automattic/wp-calypso/pull/69674/commits/738f704fcc518ca7926dfd517b50c4ea14aa9ac4.

dcalhoun commented on client/blocks/get-apps/desktop-download-card.jsx:
```diff
@@ -53,32 +61,62 @@ class DesktopDownloadCard extends Component {
+				return <SVGIcon name="linux-logo" size="24" icon={ Linux } />;
+			case WINDOWS_LINK:
+			default:
+				return <SVGIcon name="windows-logo" size="24" icon={ Windows } />;
```
It seems these icons are decorative in their use. WDYT about hiding them from screen readers? Currently, focusing on the Apple image results in merely "image" read aloud. The other icons are only used nested within anchors, so they are not focusable, but it may make sense to hide them for consistency given they do not have `title` attributes. 

For context, CSS-Tricks has a [helpful article](https://css-tricks.com/accessible-svgs/#aa-example-2-standalone-icon-decorative) on this subject. 

```suggestion
				return <SVGIcon aria-hidden="true" name="apple-logo" size="24" icon={ Apple } />;
			case 'Linux i686':
			case 'Linux i686 on x86_64':
			case LINUX_TAR_LINK:
			case LINUX_DEB_LINK:
				return <SVGIcon aria-hidden="true" name="linux-logo" size="24" icon={ Linux } />;
			case WINDOWS_LINK:
			default:
				return <SVGIcon aria-hidden="true" name="windows-logo" size="24" icon={ Windows } />;
```

dcalhoun commented on client/blocks/get-apps/mobile-download-card.jsx:
```diff
@@ -162,16 +169,84 @@ class MobileDownloadCard extends Component {
+		return (
+			<div className="get-apps__qr-code-subpanel jetpack">
+				<QRCode
+					value={ localizeUrl( 'https://apps.wordpress.com/mobile/?campaign=calypso-qrcode-apps' ) }
```
My understanding is the intention is for the QR code to direct users to the respective platform store: `https://wp.com/app`. I am uncertain as to whether the `campaign` is supported/tracked with that URL.

SiobhyB replied:
Aha, you're right. The current URL is just redirecting to the static apps.wordpress.com site on mobile, rather than the app stores. I also didn't find any results when searching our accounts for the campaign name, so your suspicion that's not supported with this current format seems right. 

I've gone ahead to update the URL (for both the WordPress and Jetpack panels) in https://github.com/Automattic/wp-calypso/pull/69674/commits/ed05e141e9689686876d499ec5041b53a82be1db. I used the `apps.wordpress.com/get?campaign=` structure that we're using in other places, which I believe is what we want.

In addition, this comment made me realise that I should have likely associated the new text link with a campaign. I've done that in https://github.com/Automattic/wp-calypso/pull/69674/commits/ecf94df2373ba79ceb4a8361a1b3a13ddaee2cf0.

Let me know if you think any other changes are needed to this and thanks for spotting this!

dcalhoun commented on client/blocks/get-apps/desktop-download-card.jsx:
```diff
@@ -153,22 +229,122 @@ class DesktopDownloadCard extends Component {
+			<>
+				<div className="get-apps__card-text jetpack">
+					<SVGIcon
+						name="desktop-app-logo"
```
Similar to my other comment, I think it makes sense to hide this from screen readers. 
```suggestion
						aria-hidden="true"
						name="desktop-app-logo"
```

SiobhyB replied:
Implemented in https://github.com/Automattic/wp-calypso/pull/69674/commits/ddb0f0c17772d96967669a39cedc2eba2d06a89e, thank you!

dcalhoun commented on client/blocks/get-apps/mobile-download-card.jsx:
```diff
@@ -162,16 +169,84 @@ class MobileDownloadCard extends Component {
+						'Everything you need to publish, manage, and grow your site anywhere, any time.'
+					) }
+				</p>
+				<AppsBadge
```
It would appear I overlooked updating the `title` for the `AppBadge` component in previous PRs. 😬 Sorry about that. Would you be willing to add a conditional text referencing "Jetpack?" It currently always says the "WordPress <iOS|Android> mobile app." 

https://github.com/Automattic/wp-calypso/blob/2053b74187dbef03cb69a9dd1fd7cbe3b34e8629/client/blocks/get-apps/apps-badge.jsx#L28

https://github.com/Automattic/wp-calypso/blob/2053b74187dbef03cb69a9dd1fd7cbe3b34e8629/client/blocks/get-apps/apps-badge.jsx#L49

SiobhyB replied:
Ah, good spot, I'd overlooked that too! Implemented that change in https://github.com/Automattic/wp-calypso/pull/69674/commits/91bdd1f2fb107b14837384399b9ad3602187b2d0.

dcalhoun commented on client/blocks/get-apps/style.scss:
```diff
@@ -7,27 +17,140 @@
+		background: #e9eff5;
+		margin-bottom: 0;
+
+		&::before {
+			content: "";
+			background-color: #e9eff5;
+			position: absolute;
+			height: 100%;
+			width: 200vw;
+			left: -100vw;
+			z-index: -1;
+		}
```
This is quite clever — I mean that in a positive way. That said, whenever we are able to remove the branding conditionals, it would likely be sensible to refactor the layout padding to make this absolutely positioned element unnecessary.

SiobhyB replied:
Thank you, I agree it'd be nice to refactor so that everything is cleaner when we go about removing the branding conditionals.

dcalhoun reviewed (commented):
This looks really great! I did not notice any issues in comparison to the original design and it functioned really well in macOS Safari, both small and large screens. 

I left a few thoughts to consider, but none are blocking.

SiobhyB committed:
Remove unnecessary fragment

SiobhyB committed:
Separate objects out for readability

Following the feedback here, this commit goes ahead to create separate objects for each conditional, rather than using mutation: https://github.com/Automattic/wp-calypso/pull/69674#discussion_r1011830979

The aim is to improve overall readability.

SiobhyB committed:
Hide decorative-only images from screen readers

Co-authored-by: David Calhoun <438664+dcalhoun@users.noreply.github.com>

SiobhyB committed:
Add Jetpack-specific title to app badges

SiobhyB committed:
Merge branch 'update/get-apps-page' of github.com:Automattic/wp-calypso into update/get-apps-page

SiobhyB committed:
Update QR code to direct mobile users to app store

The current URLs associated with the QR code always redirect to the static apps.wordpress.com site, regardless of user site. This isn't what we want. Instead, this commit updates the URLs so that they redirect mobile users to the app store.

The update in this commit will also fix the previously broken campaign tracking.

This commit follows feedback here: https://github.com/Automattic/wp-calypso/pull/69674#discussion_r1011881452

SiobhyB committed:
Enable tracking for URL

SiobhyB committed:
Hide purely decorative icon from screen readers

SiobhyB commented:
@dcalhoun, thank you for the review! I believe I've addressed all your points, but let me know if you spot anything else. 🙇‍♀️

dcalhoun commented on client/blocks/get-apps/desktop-download-card.jsx:
```diff
@@ -53,32 +61,77 @@ class DesktopDownloadCard extends Component {
-					secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
-					thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
-				};
+				return displayJetpackAppBranding
+					? {
+							firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
+							secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
+							thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
+
+							firstAvailableIcon: this.getPlatformImage( WINDOWS_LINK ),
+							secondAvailableIcon: this.getPlatformImage( LINUX_TAR_LINK ),
+							thirdAvailableIcon: this.getPlatformImage( LINUX_DEB_LINK ),
+					  }
+					: {
+							firstAvailableLink: this.getLinkAnchorTag( WINDOWS_LINK ),
+							secondAvailableLink: this.getLinkAnchorTag( LINUX_TAR_LINK ),
+							thirdAvailableLink: this.getLinkAnchorTag( LINUX_DEB_LINK ),
+					  };
```
@SiobhyB after reviewing [your latest change](https://github.com/Automattic/wp-calypso/pull/69674#discussion_r1012058709) in https://github.com/Automattic/wp-calypso/commit/59edabb39eac9a83792917b4e052316b30742863, I realize I completely misinterpreted the original code. I thought the same keys were being replaced. I realize now that we are conditionally adding keys. I think your original implementation makes total sense. Please feel free to revert https://github.com/Automattic/wp-calypso/commit/59edabb39eac9a83792917b4e052316b30742863 if you'd like. Sorry for the misguidance.

dcalhoun commented on client/blocks/get-apps/apps-badge.jsx:
```diff
@@ -25,7 +25,10 @@ const APP_STORE_BADGE_URLS = {
 		},
-		getTitleText: () => translate( 'Download the WordPress iOS mobile app.' ),
+		getTitleText: () =>
+			this.displayJetpackAppBranding
```
This line and its sibling for Android throws an error currently. Because of the structure of this configuration object, we must pass the branding flag as an argument to the function. We cannot reference a class property as done here. 

```
TypeError: undefined is not an object (evaluating 'undefined.displayJetpackAppBranding')
```

<details><summary>Patch to fix error</summary>

```diff
diff --git a/client/blocks/get-apps/apps-badge.jsx b/client/blocks/get-apps/apps-badge.jsx
index 225a473a9b..7946fd00bc 100644
--- a/client/blocks/get-apps/apps-badge.jsx
+++ b/client/blocks/get-apps/apps-badge.jsx
@@ -25,8 +25,8 @@ const APP_STORE_BADGE_URLS = {
 				: '335703880'; // WordPress
 			return `https://apps.apple.com/app/apple-store/id${ appId }?pt=299112&ct=${ utm_source }&mt=8`;
 		},
-		getTitleText: () =>
-			this.displayJetpackAppBranding
+		getTitleText: ( displayJetpackAppBranding ) =>
+			displayJetpackAppBranding
 				? translate( 'Download the Jetpack iOS mobile app.' )
 				: translate( 'Download the WordPress iOS mobile app.' ),
 		getAltText: () => translate( 'Apple App Store download badge' ),
@@ -49,8 +49,8 @@ const APP_STORE_BADGE_URLS = {
 			const appId = displayJetpackAppBranding ? 'com.jetpack.android' : 'org.wordpress.android';
 			return `https://play.google.com/store/apps/details?id=${ appId }&referrer=utm_source%3D%${ utm_source }%26utm_medium%3D${ utm_medium }%26utm_campaign%3D${ utm_campaign }`;
 		},
-		getTitleText: () =>
-			this.displayJetpackAppBranding
+		getTitleText: ( displayJetpackAppBranding ) =>
+			displayJetpackAppBranding
 				? translate( 'Download the Jetpack Android mobile app.' )
 				: translate( 'Download the WordPress Android mobile app.' ),
 		getAltText: () => translate( 'Google Play Store download badge' ),
@@ -157,7 +157,7 @@ export class AppsBadge extends PureComponent {
 				>
 					<img
 						src={ imageSrc }
-						title={ titleText ? titleText : badge.getTitleText() }
+						title={ titleText ? titleText : badge.getTitleText( this.displayJetpackAppBranding ) }
 						alt={ altText ? altText : badge.getAltText() }
 					/>
 				</a>

```

</details>

SiobhyB replied:
Oof, not sure how I managed to miss that 🤦‍♀️

Thank you for spotting so quickly! I've gone ahead to fix now.

SiobhyB committed:
Fix: Address error by properly passing prop

SiobhyB committed:
Revert "Separate objects out for readability"

This reverts commit 59edabb39eac9a83792917b4e052316b30742863.

paulforgione commented:
Hey @SiobhyB – this is looking great!

**1**. One thing I noticed is the light blue background color doesn't continue down the page if the browser window is particularly tall:

![Height](https://user-images.githubusercontent.com/4359124/199581644-dc6edb86-f53b-4604-b371-39806a32a49d.png)

**2.** I also noticed in your screenshots and on my device that the Lottie logo animation appears low resolution / blurry in Chrome. Firefox is nice and crisp though. I'm not sure if this is a browser-specific issue that we have any control over?

**3.** And lastly, the WordPress logo in the Desktop section appears to have 1 pixel being cropped off of the bottom?:
<img width="569" alt="Logo-cropped" src="https://user-images.githubusercontent.com/4359124/199584268-1dbd8c1b-0024-4b86-9b18-355e64799e26.png">


Regarding you questions:

> In the desktop app section, I used the same icon across both desktop and mobile devices, though the designs show different icons. Should I change this to use the different version of the WordPress icon for desktop (as per the design)?

The same icon for both desktop and mobile is fine, I realize now that's the icon for the desktop app, so that works!

> I know there was some discussion related to the mobile app section's copy. Should this be changed?

I'll check in on this to see where we landed.

> The font-sizes don't all precisely match up to the designs. In cases where the font-size in the designs didn't have a matching variable from this list, I went for the closest match:

Everything looks good to me, thanks for adapting. I'll make sure to reference these numbers in the future. :)

SiobhyB committed:
Fix 'hidden' 1px in desktop icon's height

SiobhyB committed:
Fix to prevent icon blurriness in Chrome

As per this issue in Lottie's GitHub repo, the animated icon can sometimes appear blurry in Chrome: https://github.com/airbnb/lottie-web/issues/2718

This commit introduces the solution provided in that GitHub issue.

SiobhyB committed:
Make bottom panel background extend larger screens

This commit ensures the bottom/desktop panel's background always extends a device's full height, even for larger screens/monitors. This s done by setting a min-height on the main container and also setting the background's height to the viewport's height.

SiobhyB commented:
Thanks so much @paulforgione! I've addressed your points below.

> 1. One thing I noticed is the light blue background color doesn't continue down the page if the browser window is particularly tall:

I've gone ahead to fix this, which you should be able to verify on the latest live branch. Let me know if any other issues come up related to the background color following the fix.

> 2. I also noticed in your screenshots and on my device that the Lottie logo animation appears low resolution / blurry in Chrome. Firefox is nice and crisp though. I'm not sure if this is a browser-specific issue that we have any control over?

I found that the blurriness was being caused by [the `translate3d()` CSS function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/translate3d) and that this has also been reported to Lottie's GitHub repo in https://github.com/airbnb/lottie-web/issues/2718. As the blurriness was very noticeable with this larger image, especially on larger monitors, I went ahead to implement the solution offered in that GitHub issue. There's some performance loss associated with this, but I think it's a worthwhile trade-off for the higher quality look of the image. I'd appreciate hearing if you have further thoughts on this, though.

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/199787780-4334188b-2c81-459b-b0c8-e2ad3ddf36cd.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/199787597-b5ee7d2f-0e85-4fb4-9753-44b324b08882.png" width="100%"> |

> 3. And lastly, the WordPress logo in the Desktop section appears to have 1 pixel being cropped off of the bottom?:

I appreciate your eagle eyes here! 🙇‍♀️ I've gone ahead to fix this now.

I think the copy is the last part that needs addressing here, but let me know if you spot any other issues too. Thanks again for your help!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7625734

Thank you @SiobhyB for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7625734).
